syntax = "proto3";

package bluetooth.hal;

import "google/protobuf/empty.proto";
import "facade/common.proto";

service HciHalFacade {
  rpc SetLoopbackMode(LoopbackModeSettings) returns (google.protobuf.Empty) {}
  rpc SendHciCommand(HciCommandPacket) returns (google.protobuf.Empty) {}
  rpc SendHciAcl(HciAclPacket) returns (google.protobuf.Empty) {}
  rpc SendHciSco(HciScoPacket) returns (google.protobuf.Empty) {}

  rpc FetchHciEvent(bluetooth.facade.EventStreamRequest) returns (stream HciEventPacket) {}
  rpc FetchHciAcl(bluetooth.facade.EventStreamRequest) returns (stream HciAclPacket) {}
  rpc FetchHciSco(bluetooth.facade.EventStreamRequest) returns (stream HciScoPacket) {}
}

message LoopbackModeSettings {
  bool enable = 1;
}

message HciEventPacket {
  bytes payload = 1;
}

message HciCommandPacket {
  bytes payload = 1;
}

message HciAclPacket {
  bytes payload = 1;
}

message HciScoPacket {
  bytes payload = 1;
}
